mypy/checkexpr.py

```diff
@@ -1606,10 +1605,13 @@ def analyze_super(self, e: SuperExpr, is_lvalue: bool) -> Type:
                         return AnyType()
                     if not self.chk.in_checked_function():
                         return AnyType()
-                    return analyze_member_access(e.name, self_type(e.info), e,
-                                                 is_lvalue, True, False,
-                                                 self.named_type, self.not_ready_callback,
-                                                 self.msg, base, chk=self.chk)
+                    selftype = self.chk.function_stack[-1].arguments[0].variable.type
```

Comment:
This is better! It's still not entirely robust. Type checking this program causes a crash:

```
class A:
    def f() -> None:
        super().x
```

Could you add a comment discussing how `selftype` and `fill_typevars(e.info)` may be different? In particular, if there is a `self: T` annotation within class `A`, `fill_typevars(e.info)` will be `A` and `selftype` will be `T`.